src/classes/BDI_DataImportService.cls

```diff
@@ -38,9 +38,14 @@
 *
 */
 global with sharing class BDI_DataImportService {
+    public static final String BATCH_NUMBER_QUERY_STR = String.join(new List<String>{
+        DataImport__c.NPSP_Data_Import_Batch__c.getDescribe().getRelationshipName(),
+        DataImportBatch__c.Batch_Number__c.getDescribe().getName()
```

Comment:
The UTIL_Describe class should probably be used here to take advantage of the caching of Describes that is done in that class.

Reply:
For the field it could use String.valueOf(DataImportBatch__c.Batch_Number__c) to avoid the describe, for the object it could get the describe from the UTIL_Describe class and then reference the relationship name.

Reply:
@kenlewis-sfdo would the goal there be to avoid the `getDescribe()` method mainly since it's a more expensive call?

Reply:
Yep.  Describe results aren't cached by the platform so NPSP cache's them in the UTIL_Describe class to try to get some of the benefits of caching.